### PR TITLE
add upgrade test for drop_index

### PR DESCRIPTION
**Pull request by robverschoor**
### Description

Add upgrade test for PR https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2216 , since procedures containing the non-supported DROP INDEX syntax could still be created in prior versions.

Signed-off-by: Rob Verschoor [rcv@amazon.com](mailto:rcv@amazon.com)

### Issues Resolved

Add upgrade test for PR https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2216 

### Test Scenarios Covered ###
* **Use case based -** N/A

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** N/A

* **Minor version upgrade tests -** Yes

* **Major version upgrade tests -** N/A

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/upgrade/13_4/schedule [hunk 1]
```diff
@@ -222,5 +222,6 @@ cast_eliminate
 TestDatatypeAggSort
 babel_index_nulls_order-before-15-5
 BABEL-2999
+drop_index-before-15_6-or-16_1
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/13_4/schedule]: Is this feature going in 16.1? I think it is meant to go in 16.2 and 15.6 so in that case the name should be `*-before-15_6-or-16_2`

File: test/JDBC/upgrade/15_5/schedule [hunk 2]
```diff
@@ -485,6 +485,7 @@ cast_eliminate
 TestDatatypeAggSort
 babel_index_nulls_order
 BABEL-2999
+drop_index-before-15_6-or-16_1
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/15_5/schedule]: I think we missed to include the original test `drop_index` in `master/schedule` file in previous PR so let's include it this time.

File: test/JDBC/expected/drop_index-before-15_6-or-16_1-vu-cleanup.out [hunk 3]
```diff
@@ -0,0 +1,11 @@
+use master
```
Comment (rishabhtanwar29) [inline test/JDBC/expected/drop_index-before-15_6-or-16_1-vu-cleanup.out]: Should cherry-pick this to BABEL4_X_DEV branch as well.

Comment (robverschoor) [reply test/JDBC/expected/drop_index-before-15_6-or-16_1-vu-cleanup.out:1]: drop_index is in test/JDBC/upgrade/latest/schedule; it was already merged before adding the upgrade tests

File: test/JDBC/upgrade/13_4/schedule (hunk 1 above)
Comment (robverschoor) [reply test/JDBC/upgrade/13_4/schedule]: I understood 16_1 but let me know if not

File: test/JDBC/upgrade/15_5/schedule (hunk 2 above)
Comment (robverschoor) [reply test/JDBC/upgrade/15_5/schedule]: drop_index is in test/JDBC/upgrade/latest/schedule; it was already merged before adding the upgrade tests

Comment (rishabhtanwar29) [reply test/JDBC/upgrade/15_5/schedule]: Correct, it is there in `latest/schedule` file but I am talking about `test/JDBC/upgrade/master/schedule` file.

File: test/JDBC/upgrade/13_4/schedule (hunk 1 above)
Comment (jsudrik) [reply test/JDBC/upgrade/13_4/schedule]: If its going in 16_1, then it should be *-before-15_5-or-16_1. If its going in 16_2, then it should be like what @rishabhtanwar29  suggests.

Comment (rishabhtanwar29) [reply test/JDBC/upgrade/13_4/schedule]: I don't see it in 16_1 but I have included it in 16_2 today. Moreover, 16.1 seems to a standalone release and actual match for 15.6 is 16.2 so let's keep it in 16.2.

Comment (robverschoor) [reply test/JDBC/upgrade/13_4/schedule]: Will correct

Comment (robverschoor) [reply test/JDBC/upgrade/13_4/schedule]: So is it 16.1 or 16.2?

File: test/JDBC/upgrade/master/schedule [hunk 4]
```diff
@@ -343,4 +343,5 @@ BABEL-4175
 BABEL_4330
 TestDatatypeAggSort
 babel_index_nulls_order
+drop_index-before-15_5-or-16_2
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/master/schedule]: This should just be `drop_index`

**Review by rishabhtanwar29 — APPROVED** [review]

File: test/JDBC/upgrade/master/schedule (hunk 4 above)
Comment (robverschoor) [reply test/JDBC/upgrade/master/schedule]: that was a separate PR, already merged earlier

**Review by jsudrik — COMMENTED** [review]
Shouldn't the name be *-before-15_6-or-16_2-* ?

**Pull request closed without merging**